apps/sms/services/js/device_storage/device_storage_client.js

```diff
@@ -0,0 +1,71 @@
+/* global bridge,
+          BroadcastChannel */
+
+(function(exports) {
+  'use strict';
+
+  /**
+   * Name of the service that is responsible for messaging functionality.
+   * @type {string}
+   */
+  const SERVICE_NAME = 'device-storage-shim';
+
+  // TODO: Remove the flag once the low storage feature is completed.
+  const LOW_STORAGE_ENABLE = false;
+
+  const priv = {
+    client: Symbol('client'),
+    appInstanceId: Symbol('appInstanceId')
+  };
+
+  var DeviceStorageClient = {
+    /**
+     * Reference to active bridge client instance.
+     * @type {Client}
+     */
+    [priv.client]: null,
+
+    /**
+     * Unique identifier of app instance where client resides in.
+     * @type {string}
+     */
+    [priv.appInstanceId]: null,
+
+    /**
+     * Initializes connection to mozMobileConnectionsShim hosted in iframe.
+     * @param {string} appInstanceId Unique identifier of app instance
+     * where client resides in.
+     */
+    init(appInstanceId) {
+      if (!appInstanceId) {
+        throw new Error('AppInstanceId is required!');
+      }
+
+      this[priv.appInstanceId] = appInstanceId;
+
+      var broadcastChannelName = `${SERVICE_NAME}-channel-${appInstanceId}`;
+
+      this[priv.client] = bridge.client({
+        service: SERVICE_NAME,
+        endpoint: new BroadcastChannel(broadcastChannelName)
+      });
+    },
+
+    /**
+     * Switch the default MMS and data to SIM that matches the given iccId.
+     * @param {string} iccId Id that could map to the target SIM.
+     * @returns {Promise} Promise of shim's method that returned when sim
+     *  switching is ready or failed.
+     */
+    on(event, func) {
+      LOW_STORAGE_ENABLE &&
+        this[priv.client].on(`${event}-${this[priv.appInstanceId]}`, func);
+    },
+
+    lowDiskSpace: LOW_STORAGE_ENABLE && navigator.getDeviceStorage('apps') &&
```

Comment:
It was defined as promise that calling the API in shim originally, but I realized that lowDiskSpace checking could happen earlier than service init ready. Even we could still do with promise return and pending the click action in inbox view, it will also make the checking in compose view much more complex. So I made an exception that called the API directly in client(I know it looks bad)... But any good suggestion is welcome here.